packages/client/src/clients/in-app-message/types.ts

```diff
@@ -0,0 +1,103 @@
+import { GenericData, PageInfo } from "@knocklabs/types";
+
+import { NetworkStatus } from "../../networkStatus";
+import { NotificationSource } from "../messages/interfaces";
+
+export interface InAppMessageTextContentField {
+  key: string;
+  rendered: string;
+  type: "text" | "markdown" | "textarea";
+  value: string;
+}
+
+export interface InAppMessageBooleanContentField {
+  key: string;
+  rendered: boolean;
+  type: "boolean";
+  value: boolean;
+}
+
+export interface InAppMessageButtonContentField {
+  key: string;
+  type: "button";
+  text: Exclude<InAppMessageTextContentField, "key | type"> & {
+    key: "text";
+    type: "text";
+  };
+  action: Exclude<InAppMessageTextContentField, "key | type"> & {
+    key: "action";
+    type: "text";
+  };
+}
+
+export type InAppMessageContentField =
+  | InAppMessageTextContentField
+  | InAppMessageBooleanContentField
+  | InAppMessageButtonContentField;
+
+export interface InAppMessage<T = GenericData> {
+  __cursor: string;
+  id: string;
+  message_type: string;
+  schema_variant: string;
+  schema_version: string;
+  content: Record<string, InAppMessageContentField>;
+  data: T | null;
+  inserted_at: string;
+  updated_at: string;
+  seen_at: string | null;
+  read_at: string | null;
+  interacted_at: string | null;
+  archived_at: string | null;
+  link_clicked_at: string | null;
+  source: NotificationSource;
+}
+
+export interface InAppMessageResponse {
+  items: InAppMessage[];
+  pageInfo: PageInfo;
+}
+
+export interface InAppMessagesQueryInfo {
```

Comment:
A big q I have is if this state needs to be hoisted here, or if it would be better served at the message-client level. What's your thinking here about why this needs to be shared?

Reply:
@cjbell Just to double check, do you mean all query info or just the duplicated data? 

I'm partial to having query data at the top level store. Having a single store is easier to work with and manage instances of. Maybe an edge case, but this also makes it so that duplicated queries update together like if you had multiple instances of a message in different parts of a page although maybe thats a stretch.